conda/notices/types.py

```diff
@@ -76,8 +77,16 @@ def _parse_iso_timestamp(iso_timestamp: Optional[str]) -> Optional[datetime]:
 
     @classmethod
     def get_cache_key(cls, url: str, name: str, cache_dir: Path) -> Path:
-        """Returns the place where this channel response will be stored as cache"""
+        """
+        Returns the place where this channel response will be stored as cache
+
+        We use a naive hashing algorithm to avoid issues with filenames (e.g. illegal or
+        problematic characters). We don't care the that this is a weak hash.
+        """
         url_obj = parse.urlparse(url)
         path = url_obj.path.replace("/", "-")
-        cache_filename = f"{name}{path}"
+        bytes_filename = bytes(f"{name}{path}", "utf-8")
+        md5_hash = hashlib.md5(bytes_filename)
```

Comment:
Unfortunately, we cannot do [binary f-strings](https://peps.python.org/pep-0498/#no-binary-f-strings) but we can use `str.encode` (where the default encoding is utf-8):

```suggestion
        md5_hash = hashlib.md5(f"{name}{path}".encode())
```

Reply:
Is there a reason why this is using md5 and not sha256?

Reply:
No, reason. Just an arbitrary choice 😅 . Like I said, super trivial to change.